pytorch_lightning/trainer/training_loop.py

```diff
@@ -703,11 +702,12 @@ def run_training_batch(self, batch, batch_idx, dataloader_idx):
                     # automatic_optimization=True: perform dpp sync only when performing optimizer_step
                     # automatic_optimization=False: don't block synchronization here
                     with self.block_ddp_sync_behaviour():
-                        self.training_step_and_backward(
+                        result = self.training_step_and_backward(
                             split_batch, batch_idx, opt_idx, optimizer, self.trainer.hiddens
                         )
 
                     batch_outputs = self._process_closure_result(
+                        opt_closure_result=result,
                         batch_outputs=batch_outputs,
                         opt_idx=opt_idx,
```

Comment:
If i'm not mistaken this batch_outputs and the one in L741-L745 can now be extracted one indentation level higher

Reply:
good observation. almost, but had to change  `self.update_running_loss(` slightly, because it assumed process_result to run first to update the accumulated loss. 

It makes it slightly harder to review here, but should help with future refactors